src/garage/tf/regressors/gaussian_mlp_regressor.py

```diff
@@ -130,60 +130,45 @@ def __init__(self,
             std_parameterization='exp',
             layer_normalization=layer_normalization)
 
+        # model for old distribution, used when trusted region is on
+        self._old_model = self.model.clone(name='model_for_old_dist')
```

Comment:
`model.clone()` does not copy the parameter. This is problematic later when calculating the KL divergence using the old model.

Reply:
There are cases where you don't want the params (e.g. because you want to create dueling Q-functions), but I think the params should be copied by default.

I think maybe you want a signature like `model.clone(reinitialize=False)`

Reply:
Good idea. I will add this in another PR.

Reply:
However in most cases, parameters are not initialized when `model.clone()` is called, before the model is not built yet. We may also have to enforce `mode.clone()` can only be called after the model is built.